src/config/settings.ts

```diff
@@ -118,6 +119,114 @@ export async function clear() {
   await fs.promises.rm(paths.config, { recursive: true, force: true } as any);
 }
 
+/** Walks the settings object given a fully-qualified accessor,
+ *  returning an updatable subtree of the settings object, along with the final subfield
+ *  in the accessor.
+ *
+ *  Clients calling this routine expect
+ *  to use it like so:
+ *   const result = getUpdatableNode({a: {b: c: {d: 1, e: 2}}}, 'a-b-c-d')
+ *   where result should be [subtree = {d: 1, e: 2}, finalFieldName = 'd']
+ *   so the caller can do `subtree[finalFieldName] = newValue`
+ *   and update that part of the preferences Config.
+ *
+ *   `result` would be null if the accessor doesn't point to a node in the Settings subtree.
+ *
+ * @param cfg: the settings object
+ * @param fqFieldAccessor: a dotted name representing a path to a node in the settings object.
+ * @returns {null} if fqFieldAccessor doesn't point to a node in the settings tree.
+ *          [internal node in cfg, final accessor name] if it does.
+ */
+export function getUpdatableNode(cfg: Settings, fqFieldAccessor: string): [RecursivePartial<Settings>, string] | null {
+  const optionParts = fqFieldAccessor.split('-');
+  const finalOptionPart = optionParts.pop() ?? '';
+
+  try {
+    const lhs = optionParts.reduce((currentConfig, currentField) => {
+      return (currentConfig as Record<string, any>)[currentField];
+    }, cfg);
+
+    if (finalOptionPart in lhs) {
+      return [lhs, finalOptionPart];
+    }
+  } catch (err: any) {
+    if (!(err instanceof TypeError)) {
+      throw err;
+    }
+    // Otherwise fqFieldAccessor doesn't point to a node in cfg and we'll return null
+  }
+
+  return null;
+}
+
+export function updateFromCommandLine(cfg: Settings, args: string[]): Settings {
+  let i = 0;
+  const lim = args.length;
+
+  // Not using a for-loop here because `i` gets incremented in the loop to handle cases of `--option value`
+  // ... in the belief that we don't usually expect the body of for-loops to manipulate the loop index.
+  while (i < lim) {
+    const arg = args[i];
+
+    if (!arg.startsWith('--')) {
+      throw new Error(`Unexpected argument '${ arg }' in command-line [${ args.join(' ') }]`);
+    }
+    const option = arg.substring(2);
+    const [fqFieldName, value] = option.split('=', 2);
+    const lhsInfo = getUpdatableNode(cfg, fqFieldName);
+
+    if (!lhsInfo) {
+      throw new Error(`Can't evaluate command-line argument ${ arg } -- no such entry in current settings at ${ join(paths.config, 'settings.json') }`);
+    }
+    const [lhs, finalFieldName] = lhsInfo;
+    const currentValue = (lhs as Record<string, any>)[finalFieldName];
+    const currentValueType = typeof (currentValue);
```

Comment:
```suggestion
    const currentValueType = typeof currentValue;
```
Silly style nit: `typeof` is not a function. (I wish it was; it'd be saner syntactically.)

Reply:
Agreed that it should be a function and not an operator....